### PR TITLE
Added concept and howto for Flink application

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
# What changed, and why it matters

Added concept information for Flink application and HowTo steps to create Flink application.

### CODE REVIEW

File: docs/products/flink/concepts/flink-applications.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+* **Source and sink tables** definition with **SQL autocomplete**: You can create source and sink tables, the SQL autocomplete feature allows you to start from a specific set of pre-filled parameters depending on the type of connector.
+* **Interactive query**: The interactive query feature allows you to validate and preview your data in the table or SQL tranformation before deploying it.
+* **Start and stop**: you can start and stop applications stop and start anytime.
+* **Automatic Versioning**: on every edit of the table or data transformation definition, a new application versions is automatically generated allowing you to keep track of the changes and revert to a previous version if necessary.
+* **Savepoints**: You can stop your application with a savepoint, allowing you to restart it from the previous state at a later time.
+* **Scalability:** You can scale your cluster depending on the demands of your specific application by parallelizing the applications into multiple tasks and executing them concurrently in a cluster.
```
Comment (ftisiot) [inline docs/products/flink/concepts/flink-applications.rst:23]: I believe we need to discuss this in more detail:
* Scalability of the cluster is about Apache Flink itself (not at application level)
* Change the level of parallelism is at application level, but needs a separate discussion

File: docs/products/flink/concepts/flink-applications.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Limitations 
+------------
+
+* **Concurrent applications:** You can create and run up to 4 concurrent applications in the Aiven for Apache Flink service.
```
Comment (ftisiot) [inline docs/products/flink/concepts/flink-applications.rst:28]: Should this be `10`?

File: docs/products/flink/concepts/flink-applications.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+* **Savepoints:** You can store up to 10 savepoints history per application deployment, allowing you to easily roll back to previous versions of your application if necessary. If you exceed this limit, you must clear the savepoints history before creating a new application version.
```
Comment (ftisiot) [inline docs/products/flink/concepts/flink-applications.rst:29]:
```suggestion
* **Savepoints:** You can store up to 10 savepoints history per application deployment, allowing you to start from a previous state if necessary. If you exceed this limit, you must clear the savepoints history before creating a new application deployment.
```

File: docs/products/flink/howto/create-flink-applications.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+=============================================
+
+:doc:`Aiven for Flink applications <../concepts/flink-applications>` in Aiven for Apache Flink® servers as a container that includes everything connected to a Flink job, including source and sink connections and data processing logic. The `Aiven Console <https://console.aiven.io/>`_ provides a guided wizard to help you build and deploy applications, simplifying the process of selecting the source and sink tables, writing data transformation statements, and validating and ingesting data through the interactive query feature.
+
+This article provides the information required to build and deploy applications on Aiven for Apache Flink service. 
+
+Create an application via the Aiven console
+--------------------------------------------
+
+Follow these steps to build your first Aiven for Flink application:
```
Comment (ftisiot) [inline docs/products/flink/howto/create-flink-applications.rst:11]: Should we have a warning that people need to create an integration first?

File: docs/products/flink/howto/create-flink-applications.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+7. Click **Next** to add the sink table, and then click **Add your first sink table**. 
+   
+   .. note::   
+    Since this is your first application, there are currently no other applications where you can import sink tables.
+    
+8.  In the **Add new sink table** screen, 
+    
+    * Select the **Integrated service** from the drop-down list. 
+    * In the **Table SQL** section, enter the statement that will be used to create the table.  
+    * Optionally, click **Run** to view how data is being pulled from the data source. This could take some time based on the data and the connection.
```
Comment (ftisiot) [inline docs/products/flink/howto/create-flink-applications.rst:37]: Not sure this is available for sink tables

File: docs/products/flink/howto/create-flink-applications.rst [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+   .. note::   
+    Since this is your first application, there are currently no other applications where you can import sink tables.
+    
+8.  In the **Add new sink table** screen, 
+    
+    * Select the **Integrated service** from the drop-down list. 
+    * In the **Table SQL** section, enter the statement that will be used to create the table.  
+    * Optionally, click **Run** to view how data is being pulled from the data source. This could take some time based on the data and the connection. 
+    * Click **Add table**. 
+9.  Click **Next** to enter the SQL statement that transforms the data from the source stream. Optionally, click **Run** to view how data is being pulled from the data source.
```
Comment (ftisiot) [inline docs/products/flink/howto/create-flink-applications.rst:39]:
```suggestion
9.  Click **Next** to enter the **SQL statement** that transforms the data from the source stream. Optionally, click **Run** to view how data is being pulled from the data source. 
```

File: docs/products/flink/howto/create-flink-applications.rst [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    .. image:: /images/products/flink/application_landingpage_view.png
+        :scale: 50 %
+        :alt: Application landing page with an view of source table, SQL statement and sink table
+    
+11. Click **Create deployment**. On the **Create new deployment** screen, 
+    
+    * Choose the version you wish to deploy. The default version for the first deployment is **Version: 1**. 
+    * Choose the savepoint from where you want to deploy. No savepoints are available for the first application deployment. 
+    * Use the toggle for **Restart on failure** to enable or disable this option.
```
Comment (ftisiot) [inline docs/products/flink/howto/create-flink-applications.rst:50]: What does this option do? We might need to explain

File: docs/products/flink/howto/create-flink-applications.rst [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    .. image:: /images/products/flink/application_landingpage_view.png
+        :scale: 50 %
+        :alt: Application landing page with an view of source table, SQL statement and sink table
+    
+11. Click **Create deployment**. On the **Create new deployment** screen, 
+    
+    * Choose the version you wish to deploy. The default version for the first deployment is **Version: 1**. 
+    * Choose the savepoint from where you want to deploy. No savepoints are available for the first application deployment. 
+    * Use the toggle for **Restart on failure** to enable or disable this option. 
+    * Enter the number of parallel instances you want to have for the task.
```
Comment (ftisiot) [inline docs/products/flink/howto/create-flink-applications.rst:51]: I believe we need to have a concept about parrallel tasks (or a reference to the Flink docs)

**Review by ftisiot — CHANGES_REQUESTED** [review]
Hi Harshini,
I made some edits and suggested others.
We might need to distinguish what is in the scope of the app (e.g. the number of parallel tasks) vs in scope of the service (e.g. scale up or down)

File: docs/products/flink/concepts/flink-applications.rst (hunk 2 above)
Comment (harshini-rangaswamy) [reply docs/products/flink/concepts/flink-applications.rst:28]: That is right. Updated this.

File: docs/products/flink/howto/create-flink-applications.rst (hunk 4 above)
Comment (harshini-rangaswamy) [reply docs/products/flink/howto/create-flink-applications.rst:11]: I wanted to add this point as a prerequisite, and it looks like I missed it. Thanks for this.

File: docs/products/flink/howto/create-flink-applications.rst (hunk 5 above)
Comment (harshini-rangaswamy) [reply docs/products/flink/howto/create-flink-applications.rst:37]: Good point. Well, the "Run" option is visible but greyed. I will update this.

File: docs/products/flink/howto/create-flink-applications.rst (hunk 8 above)
Comment (harshini-rangaswamy) [reply docs/products/flink/howto/create-flink-applications.rst:51]: Linked it to the  [Parallel task execution](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/datastream/execution/parallel/) Apache Flink doc.

File: docs/products/flink/howto/create-flink-applications.rst (hunk 7 above)
Comment (harshini-rangaswamy) [reply docs/products/flink/howto/create-flink-applications.rst:50]: Updated line to include additional info.

File: docs/products/flink/concepts/flink-applications.rst (hunk 3 above)
Comment (harshini-rangaswamy) [reply docs/products/flink/concepts/flink-applications.rst:29]: Fixed

File: docs/products/flink/howto/create-flink-applications.rst [hunk 9]
```diff
@@ -0,0 +1,63 @@
+Create an Aiven for Apache Flink® application 
+=============================================
+
+:doc:`Aiven for Flink applications <../concepts/flink-applications>` in Aiven for Apache Flink® servers as a container that includes everything connected to a Flink job, including source and sink connections and data processing logic. The `Aiven Console <https://console.aiven.io/>`_ provides a guided wizard to help you build and deploy applications, simplifying the process of selecting the source and sink tables, writing data transformation statements, and validating and ingesting data through the interactive query feature.
+
+This article provides the information required to build and deploy applications on Aiven for Apache Flink service. 
+
+Prerequisites
```
Comment (ftisiot) [inline docs/products/flink/howto/create-flink-applications.rst:8]: I wouldn't have a separate Prereq section but more a note or a warning (with a link to the relevant doc) before the list of steps to create an application.

File: docs/products/flink/concepts/flink-applications.rst (hunk 1 above)
Comment (harshini-rangaswamy) [reply docs/products/flink/concepts/flink-applications.rst:23]: Got it, thanks. Updated the text.

File: docs/products/flink/howto/create-flink-applications.rst (hunk 9 above)
Comment (harshini-rangaswamy) [reply docs/products/flink/howto/create-flink-applications.rst]: Changed this into a note.

File: _toc.yml [hunk 10]
```diff
@@ -566,6 +566,8 @@ entries:
         entries:
           - file: docs/products/flink/concepts/flink-for-analysts
           - file: docs/products/flink/concepts/flink-for-operators
+          - file: docs/products/flink/concepts/flink-applications
+            title: Flink applications
```
Comment (ftisiot) [inline _toc.yml:570]:
```suggestion
            title: Aiven for Apache Flink applications
```

**Review by ftisiot — CHANGES_REQUESTED** [review]
Same comment here, We need to refer to applications as Aiven for Apache Flink applications

**Pull request merged**